src/Processors/Formats/Impl/CSVRowInputFormat.cpp

```diff
@@ -114,6 +114,66 @@ void CSVRowInputFormat::resetParser()
     buf->reset();
 }
 
+void CSVFormatReader::skipRow()
```

Comment:
Yeah, I tried to avoid this code duplication, but couldn't make it work well. When I make the same function for both `countRows` and `fileSegmentationEngineCSVImpl` it becomes slow, and when I tried to do it using `if constexpr` the code becames quite ugly and I decided it's not worth it. Maybe I will make another attempt later